src/compiler/checker.ts

```diff
@@ -43862,6 +43862,9 @@ namespace ts {
         }
 
         function checkGrammarExpressionWithTypeArguments(node: ExpressionWithTypeArguments | TypeQueryNode) {
+            if (isExpressionWithTypeArguments(node) && isImportKeyword(node.expression) && node.typeArguments) {
+                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_cannot_have_type_arguments);

```

Comment:
Is this the right error message and fix? It seems like `import` shouldn't be parsed as an `ExpressionWithTypeArguments` at all.

Reply:
> Is this the right error message 

TS uses this error message for the following cases 
https://github.com/microsoft/TypeScript/blob/3d3fa0c0c83050d26390ef5773c471e15c8f1b21/tests/cases/conformance/dynamicImport/importCallExpressionWithTypeArgument.ts#L10-L11

TS throws the error in `checker`, I did the same for `ExpressionWithTypeArguments`

> It seems like import shouldn't be parsed as an ExpressionWithTypeArguments at all.

How should TypeScript parse `import <T>`? Without `()` this is not `ImportCallExpression`. I'm treating `import<T>` as a pre-`ImportCallExpression`, and I expect TypeScript to handle this and throw an error.

/cc @DanielRosenwasser @ahejlsberg

Reply:
I think this is the right approach. We want the parser to create a node that we can later issue a grammar error on, and `ExpressionWithTypeArguments` exactly fits the bill.

Reply:
At the very least, we can't tell a user that this is a dynamic import. How about a new message that covers both.

> This use of 'import' is invalid. 'import()' calls can be written, but they must have parentheses and cannot have type arguments.

Reply:
@DanielRosenwasser Oke.